File: CS/DXExample.Module/CodeCentralExampleInMemoryDataStoreProvider.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo.DB;
using System.Data;
using DevExpress.Xpo.DB.Helpers;

namespace DXExample.Module {
    public class CodeCentralExampleInMemoryDataStoreProvider {
        public const string XpoProviderTypeString = "CodeCentralExampleInMemoryDataSet";
        public const string ConnectionString = "XpoProvider=CodeCentralExampleInMemoryDataSet";
        private static DataSet dataSet;

        static CodeCentralExampleInMemoryDataStoreProvider() {
            try {
                dataSet = new DataSet();
                DataStoreBase.RegisterDataStoreProvider(XpoProviderTypeString, new DataStoreCreationFromStringDelegate(CreateProviderFromString));
            } catch {
                throw new Exception(string.Format("Cannot register the {0}", typeof(CodeCentralExampleInMemoryDataStoreProvider).Name));
            }
        }
        public static IDataStore CreateProviderFromString(string connectionString, AutoCreateOption autoCreateOption, out IDisposable[] objectsToDisposeOnDisconnect) {
            InMemoryDataStore result = new InMemoryDataStore(dataSet, autoCreateOption);
            objectsToDisposeOnDisconnect = new IDisposable[] { };
            return result;
        }
        public static void Register() { }
    }

}

File: VB/DXExample.Module/CodeCentralExampleInMemoryDataStoreProvider.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo.DB
Imports System.Data
Imports DevExpress.Xpo.DB.Helpers

Namespace DXExample.Module
	Public Class CodeCentralExampleInMemoryDataStoreProvider
		Public Const XpoProviderTypeString As String = "CodeCentralExampleInMemoryDataSet"
		Public Const ConnectionString As String = "XpoProvider=CodeCentralExampleInMemoryDataSet"
		Private Shared dataSet As DataSet

		Shared Sub New()
			Try
				dataSet = New DataSet()
				DataStoreBase.RegisterDataStoreProvider(XpoProviderTypeString, New DataStoreCreationFromStringDelegate(AddressOf CreateProviderFromString))
			Catch
				Throw New Exception(String.Format("Cannot register the {0}", GetType(CodeCentralExampleInMemoryDataStoreProvider).Name))
			End Try
		End Sub
		Public Shared Function CreateProviderFromString(ByVal connectionString As String, ByVal autoCreateOption As AutoCreateOption, <System.Runtime.InteropServices.Out()> ByRef objectsToDisposeOnDisconnect() As IDisposable) As IDataStore
			Dim result As New InMemoryDataStore(dataSet, autoCreateOption)
			objectsToDisposeOnDisconnect = New IDisposable() { }
			Return result
		End Function
		Public Shared Sub Register()
		End Sub
	End Class

End Namespace
